src/Utilities.js

```diff
@@ -0,0 +1,22 @@
+
+// Implementation of https://developer.mozilla.org/en-US/docs/Web/API/Element/closest
+const findClosest = (el, selector) => {
+  while ((el = el.parentElement) && !el.matches(selector)) {
```

Comment:
This one's a bit weird because the while bails if either the assignment evaluation (iirc the first inner parenthesis around `el = el.parentElement` are required) or the `matches` attempt fails. I'll add some tests for this method so we can more freely fiddle with it :) Totally open to `no-cond-assign` set to "always".